components/datadog/agent/params.go

```diff
@@ -40,18 +42,29 @@ func newParams(env *config.CommonEnvironment, options ...func(*Params) error) (*
 	if env.AgentVersion() != "" {
 		defaultVersion = WithVersion(env.AgentVersion())
 	}
-	if env.PipelineID() != "" {
-		defaultVersion = WithPipelineID(env.PipelineID())
+	versionOptions := []func(*Params) error{defaultVersion}
+
+	// If repository and/or channel are specified, force-set them
+	if env.AgentRepository() != "" {
+		versionOptions = append(versionOptions, WithRepository(env.AgentRepository()))
+	}
+	if env.AgentChannel() != "" {
+		versionOptions = append(versionOptions, WithChannel(env.AgentChannel()))
 	}
-	options = append([]func(*Params) error{defaultVersion}, options...)
+
+	// If pipeline ID is specified, force-set parameters to testing repositories
+	if env.AgentPipelineID() != "" {
+		versionOptions = append(versionOptions, WithPipelineID(env.AgentPipelineID()))
```

Comment:
It doesn't completely - in particular, the major and minor version, if set by `WithVersion`, are respected. I can make it exclusive with the `WithRepository` / `WIthChannel` settings.

Reply:
I think it is better that `WithPipelineID` overrides the other parameters.

Reply:
> I think it is better that WithPipelineID overrides the other parameters.

I don't think it should - at the minimum, it should be possible to specify the major version alongside the pipeline id, otherwise you cannot target all testing pipelines (each testing pipeline creates two channels and components, one for each Agent major version, and it should be possible to target both).